xbmc/music/windows/GUIWindowMusicBase.cpp

```diff
@@ -277,8 +279,25 @@ void CGUIWindowMusicBase::OnItemInfo(int iItem)
 
   CFileItemPtr item = m_vecItems->Get(iItem);
 
-  if (item->IsVideo())
-  { // Music video on a mixed current playlist
+  if (item->IsVideo() && item->HasVideoInfoTag() && item->GetVideoInfoTag()->m_type == "actor" &&
+      !item->GetVideoInfoTag()->m_artist.empty())
+  { // Music video artist (navigation by music > music video > artist))
+    CMusicDatabase music_database;
+    if (music_database.Open())
+    {
+      int idArtist = music_database.GetArtistByName(StringUtils::Join(
+          item->GetVideoInfoTag()->m_artist,
+          CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_videoItemSeparator));
+      if (idArtist > -1)
+        CGUIDialogMusicInfo::ShowForArtist(idArtist);
+      music_database.Close();
+    }
+    return;
+  }
+
+  if (item->IsVideo() && item->HasVideoInfoTag() &&
+      item->GetVideoInfoTag()->m_type == MediaTypeMusicVideo)
+  { // Music video on a mixed current playlist or navigation by music > music video > artist > video
```

Comment:
Although on mixed _smart_ playlists any video must be a music video,  the current music playlist can contain mix of music files and videos that are not music videos. For example when playing a .m3u that has both music and video files it can be any video file. This stops `<i>` on such a video in the current playlist from showing info as it did before.